### ISSUE TITLE
Alpine 3.10 not working with Codespaces

### ISSUE THREAD

**Issue by Chuxel** [MicrosoftDocs/vsonline#458]
Currently there is a script that is run when creating a Codespace with a custom Dockerfile that includes checking for required dependencies.  

The challenge is that the Alpine version of this script is trying to add in a package that recent versions do not include: `libssl1.0`

.NET Core now supports `libssl1.1`, so the script can be updated to accommodate working with Alpine.

Further, to improve performance, we could look to see if the packages needed are missing before attempting to install them.

//cc: @jkeech @mcolyer

**Comment by Chuxel** [comment]
It also looks like we're doing an `apk upgrade` which should not be strictly required.  I'd suggest removing it and seeing if that also helps resolve this problem.  It appears to be a really old bug with the script.

**Comment by mcolyer** [comment]
> Further, to improve performance, we could look to see if the packages needed are missing before attempting to install them.

So think the package managers are pretty fast at determining if a package is installed or not. I was more hoping we could eliminate the package index update step.

**Comment by jkeech** [comment]
> So I think the package managers are pretty fast at determining if a package is installed or not. I was more hoping we could eliminate the package index update step.

Agreed. We should first check if it is installed (maybe with a minimum version if we know what that is). And if not, then perform the update + install. I think it's still good to do the update at that point so you are getting the latest patched versions when we do the install of tools that are not built in to the image.

**Comment by danecreekphotography** [comment]
I hit this today when I was pushing myself past #536. Added bash to my container's Dockerfile (ick) and got past the bash problem then encountered this one.
